picard/formats/id3.py

```diff
@@ -650,36 +650,34 @@ def _save_tags(self, tags, filename):
             tags.update_to_v24()
             tags.save(filename, v2_version=4, v1=v1)
 
-    @property
-    def new_metadata(self):
-        config = get_config()
-        if not config.setting["write_id3v23"]:
-            return self.metadata
-
-        copy = Metadata()
-        copy.copy(self.metadata)
-
-        join_with = config.setting["id3v23_join_with"]
-        copy.multi_valued_joiner = join_with
-
-        for name, values in copy.rawitems():
-            # ID3v23 can only save TDOR dates in YYYY format. Mutagen cannot
-            # handle ID3v23 dates which are YYYY-MM rather than YYYY or
-            # YYYY-MM-DD.
-            if name == "originaldate":
-                values = [v[:4] for v in values]
-            elif name == "date":
-                values = [(v[:4] if len(v) < 10 else v) for v in values]
-
-            # If this is a multi-valued field, then it needs to be flattened,
-            # unless it's TIPL or TMCL which can still be multi-valued.
-            if (len(values) > 1 and name not in ID3File._rtipl_roles
-                    and not name.startswith("performer:")):
-                values = [join_with.join(values)]
-
-            copy[name] = values
-
-        return copy
+    def format_specific_metadata(self, metadata, tag, settings=None):
+        if not settings:
+            config = get_config()
+            settings = {
+                "write_id3v23": config.setting["write_id3v23"],
+                "id3v23_join_with": config.setting["id3v23_join_with"],
+            }
```

Comment:
Untested but such help function could simplify things if we are going to do that more often:

```python
def cached_settings(keys, config=None):
    if config is None:
        config = get_config()
    return {key: config.setting[key] for key in keys}


settings = cached_settings(["write_id3v23", "id3v23_join_with"])
```

Reply:
@zas I definitely like this idea. This does mean you need to run this code in the calling routine rather than passing an empty dict in the called routine. 

I would prefer to see the following code:
```
def cached_settings(keys, config=None, settings={}):
    if config is None:
        config = get_config()
    for key in keys:
         settings[key] = config.setting[key]
    return settings
```
which can then be used in a calling routine like this:
```
settings = cached_settings(["write_id3v23", "id3v23_join_with"])
```
or in a called routine like this:
```
fsm_settings = {}
x = format_specific_metadata(..., fsm_settings)

def format_specific_metadtata(..., settings={}):
    if not settings:
        settings = cached_settings(["write_id3v23", "id3v23_join_with"], settings)  #or
        cached_settings(["write_id3v23", "id3v23_join_with"], settings)
```